pallets/assets-info/src/lib.rs

```diff
@@ -46,8 +47,14 @@ pub trait Trait: assets::Trait {
 
     /// The maximum decimal points an asset may be.
     type MaxDecimals: Get<u32>;
+
+    type Currency: MultiTokenCurrencyExtended<Self::AccountId>;
 }
 
+type CurrencyIdOf<T> = <<T as Trait>::Currency as MultiTokenCurrency<
```

Comment:
Are there a point in recreating `CurrencyIdOf` and `BalanceOf` types in every pallet where these are used?

Reply:
it doesnt seem to be workaround for that as we are always casting to `Trait` type that is availble inside given pallet

Reply:
https://trello.com/c/OST4jO2F/106-create-single-trait-that-sets-currencyid-and-balance-and-tightly-couple-it-with-all-the-pallets-to-have-common-type-in-typescrip